gcp/observability/opencensus.go

```diff
@@ -128,9 +127,6 @@ func startOpenCensus(config *config) error {
 // packages if exporter was created.
 func stopOpenCensus() {
 	if exporter != nil {
-		internal.ClearGlobalDialOptions()
-		internal.ClearGlobalServerOptions()
```

Comment:
Why were these moved?  The global dial/server options are set in `startOpenCensus`, so it seems like we should be clearing them here (even if we need to do it unconditionally).

Reply:
Fair question. Now, global dial and server options are being added in logging.go. Unlike this file, while holds a ref to a global exporter, which can be used as an invariant to determine whether global dial options and server options have been registered in this package, logging.go has no such mechanism of determining to add a conditional check on whether they have been registered. Thus, I found it best to simply clear unconditionally in observability.go, encompassing both the cleanup for logging (which we can't do conditionally), and also for openCensus (which we can do conditionally.

Reply:
We could always clear it in both places.  It shouldn't be an error to clear it if it's already clear.

Minor point, but it's nice if `stopSomething` undoes `startSomething` instead of relying on adjacent code.

Reply:
Discussed offline; left opencensus.go as is and cleared within the exporter != nil invariant, and logging.go cleared server/dial options unconditionally.